### PR TITLE
FEAT: ABI and network static library .a / .lib

### PR DESCRIPTION
**Pull request by mathieurobert1**
With @AxelF44, we created a static library r-type_abi_network to handle Network in the project.

With that library, in the server or the client you just have to target_include the library and used the ABIClient.hpp or ABIServer.hpp located in the folder Network/ABI at the root of the project.
image

For the moment, there are just hpp files that are complete. All cpp files will be created in an other issue.
Same for the compilation, only the empty file Networ/ABI/ABINetwork.cpp is compiled.

### CODE REVIEW

File: Network/ABI/ABIClient.hpp
Comment (Peralban) [inline Network/ABI/ABIClient.hpp:1]: Add comment

**Review by Peralban — CHANGES_REQUESTED** [review]
Put docs

**Review by Njord201 — APPROVED** [review]

**Review by Njord201 — CHANGES_REQUESTED** [review]
I'm disappointed Mathieu, a good dev doesn't forget the basic documentation.

**Review by Peralban — CHANGES_REQUESTED** [review]
Make à ABI error class

File: Network/Errors/IError.hpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+         * @return const char* Error message
+         */
+        virtual const char *what() const noexcept = 0;
+};
+
+/**
+ * @brief Abstract class for error handling
+ *
+ */
+class AError : public IError {
```
Comment (Njord201) [inline Network/Errors/IError.hpp:39]: A AError in IError ?

**Review by Peralban — APPROVED** [review]

**Review by AxelF44 — APPROVED** [review]

**Pull request merged**